# Analyzing - Looking Into How Parks in Paradise Compare to Cities of Similar Size

In [1]:
processed_outdir = '/Users/kavishharjai/code/mh_paradise/processed/'

In [2]:
import pandas as pd 
import numpy as np
import os
import altair as alt
#import pandas and numpy libraries

In [3]:
pd.set_option('display.max_rows', 500)

In [5]:
merged = pd.read_csv(processed_outdir + 'merged.csv')
merged

Unnamed: 0                             city  popestimate2018  mh_spaces
0             0                         Adelanto            34411      257.0
1             1                     Agoura Hills            20307        NaN
2             2                          Alameda            80196        NaN
3             3                           Albany            20352        NaN
4             4                         Alhambra            84666        NaN
5             5                      Aliso Viejo            50739        NaN
6             6                          Alturas             2530       67.0
7             7                      Amador City              196        NaN
8             8                  American Canyon            20412      843.0
9             9                          Anaheim           353043     3542.0
10           10                         Anderson            10439      691.0
11           11                           Angels             3863        NaN
12           12                          Antioch           111622      379.0
13           13                     Apple Valley            73521     1427.0
14           14                          Arcadia            57105        NaN
15           15                           Arcata            18374      618.0
16           16                    Arroyo Grande            17970      889.0
17           17                          Artesia            16488       97.0
18           18                            Arvin            21275      228.0
19           19                       Atascadero            30549      398.0
20           20                         Atherton             7083        NaN
21           21                          Atwater            30551      628.0
22           22                           Auburn            14126     1006.0
23           23                           Avalon             3747       17.0
24           24                           Avenal            13310       23.0
25           25                            Azusa            49771      588.0
26           26                      Bakersfield           379915     4665.0
27           27                     Baldwin Park            75807      275.0
28           28                          Banning            30363      981.0
29           29                          Barstow            23564     1296.0
30           30                         Beaumont            48297      387.0
31           31                             Bell            35348       80.0
32           32                       Bellflower            76439     1423.0
33           33                     Bell Gardens            41874      412.0
34           34                          Belmont            27071       91.0
35           35                        Belvedere             2128        NaN
36           36                          Benicia            28219      266.0
37           37                         Berkeley           123541        NaN
38           38                    Beverly Hills            33808        NaN
39           39                    Big Bear Lake             5305      298.0
40           40                            Biggs             1920        2.0
41           41                           Bishop             3741      994.0
42           42                        Blue Lake             1259       41.0
43           43                           Blythe            19943      459.0
44           44                         Bradbury             1058        NaN
45           45                          Brawley            25935      344.0
46           46                             Brea            44170      952.0
47           47                        Brentwood            63435      245.0
48           48                         Brisbane             4667       61.0
49           49                         Buellton             5328      308.0
50           50                       Buena Park            82417      256.0
5

In [6]:
merged[merged.city == 'Paradise']

Unnamed: 0      city  popestimate2018  mh_spaces
309         309  Paradise            26532     1586.0

The population of Paradise in 2018 was in the high 20,000s. I will find cities that have populations within two different ranges 20k to 30k and 25k to 35k.

In [7]:
twenty_to_thirty = merged[(merged['popestimate2018']>= 20000) & (merged['popestimate2018'] <= 30000)]


In [8]:
twenty_to_thirty.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 57 entries, 1 to 481
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Unnamed: 0       57 non-null     int64  
 1   city             57 non-null     object 
 2   popestimate2018  57 non-null     int64  
 3   mh_spaces        42 non-null     float64
dtypes: float64(1), int64(2), object(1)
memory usage: 2.2+ KB


In [9]:
twenty_to_thirty = twenty_to_thirty.dropna()

In [10]:
sort_spaces_twenty_to_thirty = twenty_to_thirty.sort_values('mh_spaces', ascending= False)

In [11]:
sort_spaces_twenty_to_thirty

Unnamed: 0                city  popestimate2018  mh_spaces
110         110  Desert Hot Springs            29449     3277.0
309         309            Paradise            26532     1586.0
29           29             Barstow            23564     1296.0
346         346          Ridgecrest            28964     1078.0
8             8     American Canyon            20412      843.0
481         481        Yucca Valley            21730      772.0
134         134              Eureka            27009      768.0
418         418    South Lake Tahoe            22700      678.0
442         442    Twentynine Palms            26769      568.0
232         232          Loma Linda            24194      530.0
233         233              Lomita            20279      525.0
290         290             Oakdale            23081      479.0
471         471             Windsor            27979      437.0
99           99              Cudahy            23655      415.0
252         252              Marina            22203      399.0
225         225             Lemoore            25809      392.0
374         374              Sanger            26358      380.0
405         405               Selma            24438      366.0
45           45             Brawley            25935      344.0
153         153                Galt            26343      343.0
221         221             Lathrop            23713      333.0
112         112              Dinuba            24368      301.0
36           36             Benicia            28219      266.0
350         350           Riverbank            24637      247.0
18           18               Arvin            21275      228.0
53           53           Calabasas            23826      224.0
412         412             Soledad            25812      186.0
204         204        Laguna Beach            23018      185.0
343         343             Reedley            25801      179.0
120         120      East Palo Alto            29625      146.0
117         117              Duarte            21413      141.0
453         453              Walnut            29882      133.0
313         313           Patterson            22484      128.0
402         402          Seal Beach            24263      125.0
90           90            Corcoran            21801      120.0
455         455               Wasco            27533      107.0
34           34             Belmont            27071       91.0
188         188      Imperial Beach            27317       81.0
371         371        San Fernando            24148       76.0
423         423         Suisun City            29561       54.0
224         224         Lemon Grove            26782       44.0
113         113               Dixon            20203       34.0

In [29]:
twentyf_to_thirtyf = merged[(merged['popestimate2018']>= 25000) & (merged['popestimate2018'] <= 35000)]
twentyf_to_thirtyf.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 49 entries, 0 to 471
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Unnamed: 0       49 non-null     int64  
 1   city             49 non-null     object 
 2   popestimate2018  49 non-null     int64  
 3   mh_spaces        35 non-null     float64
dtypes: float64(1), int64(2), object(1)
memory usage: 1.9+ KB


In [30]:
twentyf_to_thirtyf = twentyf_to_thirtyf.dropna()

In [31]:
sort_spaces_twentyf_to_thirtyf = twentyf_to_thirtyf.sort_values('mh_spaces', ascending= False)

In [32]:
sort_spaces_twentyf_to_thirtyf

Unnamed: 0                city  popestimate2018  mh_spaces
110         110  Desert Hot Springs            29449     3277.0
222         222            La Verne            32324     1754.0
309         309            Paradise            26532     1586.0
346         346          Ridgecrest            28964     1078.0
28           28             Banning            30363      981.0
370         370           San Dimas            33888      906.0
396         396         Santa Paula            30298      775.0
134         134              Eureka            27009      768.0
385         385           San Pablo            31050      718.0
159         159              Goleta            32087      650.0
21           21             Atwater            30551      628.0
442         442    Twentynine Palms            26769      568.0
403         403             Seaside            33976      488.0
471         471             Windsor            27979      437.0
19           19          Atascadero            30549      398.0
225         225             Lemoore            25809      392.0
374         374              Sanger            26358      380.0
45           45             Brawley            25935      344.0
153         153                Galt            26343      343.0
36           36             Benicia            28219      266.0
0             0            Adelanto            34411      257.0
205         205        Laguna Hills            31690      252.0
412         412             Soledad            25812      186.0
343         343             Reedley            25801      179.0
104         104          Dana Point            33804      169.0
120         120      East Palo Alto            29625      146.0
242         242           Los Gatos            30924      139.0
453         453              Walnut            29882      133.0
223         223            Lawndale            32570      110.0
455         455               Wasco            27533      107.0
34           34             Belmont            27071       91.0
188         188      Imperial Beach            27317       81.0
423         423         Suisun City            29561       54.0
224         224         Lemon Grove            26782       44.0
323         323       Pleasant Hill            34977       43.0

### Conclusions

* Paradise's population before the fire was ~26,532.
* Paradise, compared to other similarly populated cities in 2018, had a high number of MH lots.
* When compared to cities that had a population of 20,000 to 30,000, Paradise had the second highest number of MH lots. 
* When compared to cities that had a population fo 25,000 to 35,000, Paradise had the third highest number of MH lots.
* Potential sentence for thesis:
    
    **The town of Paradise had one of the highest number of mobile home park lots compared to cities that had a similar population in 2018.** 